content/ch-quantum-hardware/ac-Stark-shift.ipynb

```diff
@@ -0,0 +1,465 @@
+      "text/plain": [
+       "<Figure size 432x288 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "show_individual_traces=False\n",
+    "skip_idx=3  # number of points to skip\n",
+    "\n",
+    "center=[]\n",
+    "fwhm=[]\n",
+    "for i in range(len(resonator_tone_values)):\n",
+    "    popt,pcov=fit_gaussian(freqs,np.abs(np.real(resonator_tone_values[i])))\n",
+    "    center.append(popt[0])\n",
+    "    fwhm.append(2.355*popt[1])\n",
+    "    if show_individual_traces:\n",
+    "        plt.plot(freqs/1e3, np.real(resonator_tone_values[i]))\n",
+    "        plt.plot(freqs/1e3, gaussian(freqs,*popt), '--')\n",
+    "if show_individual_traces: plt.show()\n",
+    "\n",
+    "center_fit=np.polyfit(resonator_tone_amplitude[skip_idx:], (center[skip_idx:]-center[0]),1)\n",
+    "plt.plot(resonator_tone_amplitude[skip_idx:], np.poly1d(center_fit/1e6)(resonator_tone_amplitude[skip_idx:]), '--', lw=2, color='grey')\n",
+    "plt.plot(resonator_tone_amplitude[skip_idx:], (center[skip_idx:]-center[0])/1e6, 'o', color='black')\n",
+    "plt.xlabel(r'Resonator tone amplitude [V]')\n",
+    "plt.ylabel(r'$\\delta \\omega_q (MHz)$')\n",
+    "plt.show()\n",
+    "\n",
+    "fwhm_fit=np.polyfit(resonator_tone_amplitude[skip_idx:], np.array(fwhm[skip_idx:]),1)\n",
+    "plt.plot(resonator_tone_amplitude[skip_idx:], np.poly1d(fwhm_fit/1e6)(resonator_tone_amplitude[skip_idx:]), '--', lw=2, color='orange')\n",
+    "plt.plot(resonator_tone_amplitude[skip_idx:], np.array(fwhm[skip_idx:])/1e6, 'o', color='red')\n",
+    "plt.xlabel(r'Resonator tone amplitude [V]')\n",
+    "plt.ylabel(r'FWHM (MHz)')\n",
+    "plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### In this chapter we discuss the ac-Stark shift that the qubit experiences due to the presence of photons in the resonator. We use Qiskit Pulse to measure the qubit frequency shift and linewidth broadening"
```

Comment:
Same here, I think this is meant for the introduction

Reply:
We do also mention this in the introduction. It'd be nice to keep this text in the conclusion